src/common/com/intellij/plugins/haxe/lang/psi/impl/AbstractHaxePsiClass.java

```diff
@@ -252,7 +252,11 @@ public boolean isGeneric() {
 
   @Override
   public boolean isEnum() {
-    return (HaxeComponentType.typeOf(this) == HaxeComponentType.ENUM) || this.isAbstract() && hasMeta("@:enum");
+    if (HaxeComponentType.typeOf(this) == HaxeComponentType.ENUM) return true;
+    if (isAbstract()) {
+      return hasMeta("@:enum") || ((HaxeAbstractClassDeclaration)this).getAbstractClassType().getFirstChild().getText().equals("enum");
```

Comment:
There's got to be a better way than encoding the string "enum".  Perhaps if we make 'enum' and 'abstract' their own tokens?  I don't know.  I'd like it to read better, but I don't want to make more work (for you).

Maybe as a follow-on enhancement (new issue), we can make it so that HaxePsiToken doesn't exist.  Particularly for keywords.  What do you think?

Reply:
Good idea. But there will be a lot of work if we'll switch to explicit tokens. Good intention for the future enhancements.